ods_ci/tests/Resources/Page/ODH/Prometheus/Prometheus.robot

```diff
@@ -335,3 +335,26 @@ Run Query Range
     ...    headers=${pm_headers}    verify=${False}
     Request Should Be Successful
     RETURN    ${resp}
+
+Set Prometheus Variables
+    [Documentation]    Update Global Variables RHODS_PROMETHEUS_URL and RHODS_PROMETHEUS_TOKEN if missing
+    ${result} 	Run Process 	oc get route -l app\=prometheus -A -o jsonpath\='{.items[0].metadata.namespace}'    shell=yes
+    IF    ${result.rc} == 0
+        ${prom_namespace}=    Set Variable    ${result.stdout}
+        Log    Prometheus is running in the namespace: ${prom_namespace}    console=True
+        IF    "${RHODS_PROMETHEUS_URL}" == "" or "${RHODS_PROMETHEUS_URL}" == "PROMETHEUS_URL"
+            ${result} 	Run Process 	oc get route prometheus -n ${prom_namespace} -o jsonpath\='{.spec.host}'    shell=yes
+            Should Be True    ${result.rc} == 0    Error getting Prometheus host: ${result.stderr}
+            ${RHODS_PROMETHEUS_URL}=    Set Variable    https://${result.stdout}
+            Log    Prometheus URL: ${RHODS_PROMETHEUS_URL}    console=True
+            Set Global Variable   ${RHODS_PROMETHEUS_URL}
+        END
+        IF    "${RHODS_PROMETHEUS_TOKEN}" == "" or "${RHODS_PROMETHEUS_TOKEN}" == "PROMETHEUS_TOKEN"
+            ${result} 	Run Process 	oc create token prometheus -n ${prom_namespace} --duration 6h    shell=yes
+            Should Be True    ${result.rc} == 0    Error creating Prometheus token: ${result.stderr}
+            ${RHODS_PROMETHEUS_TOKEN}=    Set Variable    ${result.stdout}
+            Set Global Variable   ${RHODS_PROMETHEUS_TOKEN}
+        END
+    ELSE
+        Log    message=No Prometheus namespace found   level=WARN
+    END
```

Comment:
could you please move the lines about the token into a separate kw and set the kw as Protected Keyword in the RHOSi.resource?